tests: posix: headers: build error with minimal libc

**Describe the bug**

The `tests/posix/headers/portability.posix.headers.without_posix_api` test fails to build in case of toolchain which uses minimal libc by default (for example ARC MWDT toolchain):

```
In file included from /verification_workspace/zephyr/tests/posix/headers/src/aio_h.c:12:
/verification_workspace/zephyr/include/zephyr/posix/aio.h:24:18: error: field has incomplete type 'struct sigevent'
        struct sigevent aio_sigevent;
                        ^
/verification_workspace/zephyr/include/zephyr/posix/aio.h:24:9: note: forward declaration of 'struct sigevent'
        struct sigevent aio_sigevent;
               ^
1 error generated.
ninja: build stopped: subcommand failed.
```

**Expected behavior**
No build error should happen

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain: MWDT 2024.06
 - Commit SHA: recent main, 1ed04e7b49c5bec8e09f81480b7f95ae51b59b10

--- a/lib/libc/minimal/include/signal.h
+++ b/lib/libc/minimal/include/signal.h
@@ -1,0 +1,12 @@
+/*
+ * Copyright (c) 2024 Synopsys
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_LIB_LIBC_MINIMAL_INCLUDE_SIGNAL_H_
+#define ZEPHYR_LIB_LIBC_MINIMAL_INCLUDE_SIGNAL_H_
+
+#include <zephyr/posix/signal.h>
+
+#endif /* ZEPHYR_LIB_LIBC_MINIMAL_INCLUDE_SIGNAL_H_ */

